### PR TITLE
Rename task file names

### LINKED ISSUE #215: Change task.json name pattern
**Is your feature request related to a problem? Please describe.**
The current naming of task configuration files is arbitrary.

**Describe the solution you'd like**
We should make some rules about naming task.json. For example, "task_<task type>_<id>.json"

### PR DESCRIPTION
**Pull request by xingjian-zhang**
This PR renames the task files.

## Description
- Rename task files
- Remove `requirements.txt` since we have `setup.py`

```
datasets/NELL-995/task_predict_relation.json            ->      datasets/NELL-995/task_kg_relation_predication_1.json
datasets/NELL-995/task_predict_entity.json              ->      datasets/NELL-995/task_kg_entity_prediction_1.json
datasets/arxiv-year/task.json                           ->      datasets/arxiv-year/task_node_classification_1.json
datasets/cornell/task.json                              ->      datasets/cornell/task_node_classification_1.json
datasets/ogbg-molpcba/ogbg-molpcba_task.json            ->      datasets/ogbg-molpcba/task_graph_classification_1.json
datasets/ogbg-molhiv/task.json                          ->      datasets/ogbg-molhiv/task_graph_classification_1.json
datasets/chameleon/task.json                            ->      datasets/chameleon/task_node_classification_1.json
datasets/FB13/task_predict_relation.json                ->      datasets/FB13/task_kg_relation_predication_1.json
datasets/FB13/task_predict_entity.json                  ->      datasets/FB13/task_kg_entity_prediction_1.json
datasets/citeseer/task.json                             ->      datasets/citeseer/task_node_classification_1.json
datasets/FB15K237/task_predict_relation.json            ->      datasets/FB15K237/task_kg_relation_predication_1.json
datasets/FB15K237/task_predict_entity.json              ->      datasets/FB15K237/task_kg_entity_prediction_1.json
datasets/penn94/task.json                               ->      datasets/penn94/task_node_classification_1.json
datasets/ogbg-molfreesolv/ogbg-molfreesolv_task.json    ->      datasets/ogbg-molfreesolv/task_graph_regression_1.json
datasets/wiki/task.json                                 ->      datasets/wiki/task_node_classification_1.json
datasets/wisconsin/task.json                            ->      datasets/wisconsin/task_node_classification_1.json
datasets/ogbn-arxiv/ogbn-arxiv_task.json                ->      datasets/ogbn-arxiv/task_node_classification_1.json
datasets/WN18RR/task_predict_relation.json              ->      datasets/WN18RR/task_kg_relation_predication_1.json
datasets/WN18RR/task_predict_entity.json                ->      datasets/WN18RR/task_kg_entity_prediction_1.json
datasets/cora/task.json                                 ->      datasets/cora/task_node_classification_1.json
datasets/snap-patents/task.json                         ->      datasets/snap-patents/task_node_classification_1.json
datasets/pokec/task.json                                ->      datasets/pokec/task_node_classification_1.json
datasets/ogbn-products/ogbn-products_task.json          ->      datasets/ogbn-products/task_node_classification_1.json
datasets/texas/task.json                                ->      datasets/texas/task_node_classification_1.json
datasets/ogbn-mag/task.json                             ->      datasets/ogbn-mag/task_node_classification_1.json
datasets/pubmed/task.json                               ->      datasets/pubmed/task_node_classification_1.json
datasets/ogbl-collab/task_runtime_sampling.json         ->      datasets/ogbl-collab/task_time_dependent_link_prediction_1.json
datasets/ogbl-collab/task_prestore_neg.json             ->      datasets/ogbl-collab/task_time_dependent_link_prediction_2.json
datasets/twitch-gamers/task.json                        ->      datasets/twitch-gamers/task_node_classification_1.json
datasets/ogbn-proteins/ogbn-proteins_task.json          ->      datasets/ogbn-proteins/task_node_classification_1.json
datasets/actor/task.json                                ->      datasets/actor/task_node_classification_1.json
datasets/squirrel/task.json                             ->      datasets/squirrel/task_node_classification_1.json
datasets/YAGO3-10/task_predict_relation.json            ->      datasets/YAGO3-10/task_kg_relation_predication_1.json
datasets/YAGO3-10/task_predict_entity.json              ->      datasets/YAGO3-10/task_kg_entity_prediction_1.json
datasets/WN18/task_predict_relation.json                ->      datasets/WN18/task_kg_relation_predication_1.json
datasets/WN18/task_predict_entity.json                  ->      datasets/WN18/task_kg_entity_prediction_1.json
datasets/WN11/task_predict_relation.json                ->      datasets/WN11/task_kg_relation_predication_1.json
datasets/WN11/task_predict_entity.json                  ->      datasets/WN11/task_kg_entity_prediction_1.json
datasets/genius/task.json                               ->      datasets/genius/task_node_classification_1.json
datasets/FB15K/task_predict_relation.json               ->      datasets/FB15K/task_kg_relation_predication_1.json
datasets/FB15K/task_predict_entity.json                 ->      datasets/FB15K/task_kg_entity_prediction_1.json

```

## Related Issue

Fixed #215 

## Motivation and Context

The rename pattern is:
`task_<task_type>_<id>.json`.
- `<task_type>` is in snakecase.
- `<id>` starts from 1.

The script that renames all task files:
```python
import os
import json


name_map = {
    "NodeClassification": "node_classification",
    "GraphClassification": "graph_classification",
    "LinkPrediction": "link_prediction",
    "TimeDependentLinkPrediction": "time_dependent_link_prediction",
    "KGRelationPrediction": "kg_relation_predication",
    "KGEntityPrediction": "kg_entity_prediction",
    "GraphRegression": "graph_regression",
    "NodeRegression": "node_regression"
}

FOLDER = "datasets"
for dataset in os.listdir(FOLDER):
    dataset = os.path.join(FOLDER, dataset)
    if os.path.isdir(dataset):
        files = [_ for _ in os.listdir(dataset) if _.endswith(".json")]
        files.remove('metadata.json')
        files.remove('urls.json')
        stats = {k: 1 for k in name_map}
        for f in files:
            assert "task" in f
            old_task_file_path = os.path.join(dataset, f)
            with open(old_task_file_path, "r") as fp:
                task_config = json.load(fp)
                task_type = task_config["type"]
                new_name = "task_"
                new_name += name_map[task_type]
                new_name += f"_{stats[task_type]}"
                new_name += ".json"
                stats[task_type] += 1

            new_task_file_path = os.path.join(dataset, new_name)
            os.rename(old_task_file_path, new_task_file_path)
```

## How Has This Been Tested?
See GitHub Actions

### CODE REVIEW

**Comment by xingjian-zhang** [comment]
Currently, the function to get a dataset is
```python
>>> gli.get_gli_dataset(dataset="cora", task="task_node_classification_1")
```
Do you think it is better to change it to something like
```python
>>> gli.get_gli_dataset(dataset="cora", task="NodeClassification", task_id=1)

```
@jiaqima

**Comment by jiaqima** [comment]
> Currently, the function to get a dataset is
> 
> ```python
> >>> gli.get_gli_dataset(dataset="cora", task="task_node_classification_1")
> ```
> 
> Do you think it is better to change it to something like
> 
> ```python
> >>> gli.get_gli_dataset(dataset="cora", task="NodeClassification", task_id=1)
> ```
> 
> @jiaqima

Yes, the latter looks better. And we can set task_id=1 as default value.

**Comment by xingjian-zhang** [comment]
The testing failed at https://github.com/Graph-Learning-Benchmarks/gli/runs/8060149708?check_suite_focus=true#step:7:64 but does not have proper error messages. Is this due to OOM?

**Comment by xingjian-zhang** [comment]
@jiaqima Some testings failed because we changed the task names. So the testing needs to update, too.

File: gli/dataloading.py [hunk 1]
```diff
@@ -76,19 +80,55 @@ def get_gli_graph(dataset: str, device="cpu", verbose=True):
...
+        "TimeDependentLinkPrediction": "time_dependent_link_prediction",
+        "KGRelationPrediction": "kg_relation_predication",
+        "KGEntityPrediction": "kg_entity_prediction",
+        "GraphRegression": "graph_regression",
+        "NodeRegression": "node_regression"
+    }
     data_dir = os.path.join(ROOT_PATH, "datasets/", dataset)
-    task_path = os.path.join(data_dir, f"{task}.json")
+    task_path = os.path.join(data_dir, f"task_{name_map[task]}_{task_id}.json")
+    if task not in name_map:
```
Comment (jiaqima) [inline gli/dataloading.py:130]: Make the error check before line 129.

**Comment by jiaqima** [comment]
> @jiaqima Some testings failed because we changed the task names. So the testing needs to update, too.

I don't understand why this is the case. Before we change the argument API of `get_gli_dataset`, the test should work for whatever task json files as long as they start with "task_" and end with ".json".

Likely it's an out-of-memory error: https://stackoverflow.com/questions/43268156/process-finished-with-exit-code-137-in-pycharm

**Comment by xingjian-zhang** [comment]
Since we are using the new function signatures, users must pass task type like `NodeClassiciation` rather than task config file name like `task` as before. Therefore, in our testing script, testers need to provide the task types.

**Comment by jiaqima** [comment]
> Since we are using the new function signatures, users must pass task type like `NodeClassiciation` rather than task config file name like `task` as before. Therefore, in our testing script, testers need to provide the task types.

I understand that test needs to be changed after your commit [6c7baab](https://github.com/Graph-Learning-Benchmarks/gli/pull/255/commits/6c7baabb89c3081c63254f4d644f9372a0f5450d). But test should not break before that commit.

**Comment by xingjian-zhang** [comment]
Hi @jiaqima, I rollback the version to [`cc4c314` (#255)](https://github.com/Graph-Learning-Benchmarks/gli/pull/255/commits/cc4c3141d70aa53a83a7ae1a6bcdeabbb9b98815) and had a test. It seems that only `ogbg-molpcba` has an error (#259) due to a missing attribute. All other dataloading tests passed smoothly:
image

**Comment by StevenGolden1203** [comment]
I have identified the bug in `test_data_loading.py`. I will fix them in this pr after permission is given

**Comment by xingjian-zhang** [comment]
Hi @StevenGolden1203, it seems that you made some commits on my developing branch directly. Could you please open another branch if you want to make any changes?

**Comment by jiaqima** [comment]
> Hi @StevenGolden1203, it seems that you made some commits on my developing branch directly. Could you please open another branch if you want to make any changes?

I asked @StevenGolden1203 to help fix the test code on Monday.

**Comment by StevenGolden1203** [comment]
Two types of error remains.
1. `ogbg-molfreesolv` and `ogbg-molpcba` fails `test_data_loading.py`, which is mentioned in https://github.com/Graph-Learning-Benchmarks/gli/issues/263 and https://github.com/Graph-Learning-Benchmarks/gli/issues/258.

2. `test_kg_training.py` raises some `NotImplementedError`. Maybe we can change the Error into Warning, in order not to block the test here? @jiaqima 

image

**Comment by StevenGolden1203** [comment]
Hi @xingjian-zhang, sorry to block your workflow. I think I need one more commit to fix `test_kg_training.py` here.

**Comment by jiaqima** [comment]
> 2. `test_kg_training.py` raises some `NotImplementedError`. Maybe we can change the Error into Warning, in order not to block the test here? @jiaqima 

I think the errors come from these lines:

https://github.com/Graph-Learning-Benchmarks/gli/blob/dd98d4e37cce200aa53a001aeb96137b5e0b00c2/tests/test_kg_training.py#L29

https://github.com/Graph-Learning-Benchmarks/gli/blob/dd98d4e37cce200aa53a001aeb96137b5e0b00c2/tests/test_kg_training.py#L107

Could you change the arguments here as well?

**Comment by StevenGolden1203** [comment]
@jiaqima Yes, I think I wrongly identified the bugs in above comments. I will fix `test_kg_training.py` now

**Comment by xingjian-zhang** [comment]
No problem!

**Comment by StevenGolden1203** [comment]
After https://github.com/Graph-Learning-Benchmarks/gli/issues/258 is fixed, I think this PR can pass all the tests.

image

**Comment by StevenGolden1203** [comment]
@xingjian-zhang Can you merge `main` into `setup` please, so we can see if https://github.com/Graph-Learning-Benchmarks/gli/issues/258 is fixed. Much thanks!

**Comment by jiaqima** [comment]
> @xingjian-zhang Can you merge `main` into `setup` please, so we can see if #258 is fixed. Much thanks!

The test passed. I'm merging it now.

**Pull request merged**